From 8e64d683f2fc4ba9e79bfd0c66a603877d7e94eb Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Mon, 1 Jul 2019 20:19:58 -0400
Subject: [PATCH] arm64: make CONFIG_ARCH_BCM2835 select CONFIG_MFD_CORE

The bcm2835-pm.c driver is automatically selected as a built-in by
ARCH_BCM2835, and its probe function calls devm_mfd_add_devices, which
comes from MFD_CORE.

A missing Kconfig dependency/selection means it's possible to select
CONFIG_MFD_CORE=m with CONFIG_ARCH_BCM2835=y, leading to build failures.

Fix by making CONFIG_ARCH_BCM2835 select CONFIG_MFD_CORE, which will
pull in the MFD core as a built-in rather than a module.
---
 arch/arm64/Kconfig.platforms | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/Kconfig.platforms b/arch/arm64/Kconfig.platforms
index e9bdec0ead72..38d8aa29b111 100644
--- a/arch/arm64/Kconfig.platforms
+++ b/arch/arm64/Kconfig.platforms
@@ -25,6 +25,7 @@ config ARCH_BCM2835
 	select ARM_AMBA
 	select ARM_TIMER_SP804
 	select HAVE_ARM_ARCH_TIMER
+	select MFD_CORE
 	help
 	  This enables support for the Broadcom BCM2837 SoC.
 	  This SoC is used in the Raspberry Pi 3 device.
-- 
2.22.0

